rest/model/user.go

```diff
@@ -26,3 +27,108 @@ func (apiPubKey *APIPubKey) BuildFromService(h interface{}) error {
 func (apiPubKey *APIPubKey) ToService() (interface{}, error) {
 	return nil, errors.Errorf("ToService() is not impelemented for APIPubKey")
 }
+
+type APIUserSettings struct {
+	Timezone      APIString                  `json:"timezone"`
+	GithubUser    APIGithubUser              `json:"github_user"`
+	SlackUsername APIString                  `json:"slack_username"`
+	Notifications APINotificationPreferences `json:"notifications"`
+}
+
+func (s *APIUserSettings) BuildFromService(h interface{}) error {
+	switch v := h.(type) {
+	case user.UserSettings:
+		s.Timezone = ToAPIString(v.Timezone)
+		s.SlackUsername = ToAPIString(v.SlackUsername)
+		err := s.GithubUser.BuildFromService(v.GithubUser)
```

Comment:
I was imagining that GithubUser could be empty, if the user hadn't set it yet. Does this noerror in that case?

Reply:
If that property is empty, this will also return an empty GithubUser